Source/DafnyRuntime/DafnyRuntime.py

```diff
@@ -39,10 +39,31 @@ def _tail_call():
     raise TailCall()
 
 class Seq(list):
+    def __init__(self, __iterable = None, isStr = False):
+        if __iterable is None:
+            __iterable = []
+        self.isStr = isinstance(__iterable, str) or isStr
+        super().__init__(__iterable)
+
     @property
     def Elements(self):
         return self
 
+    @property
+    def UniqueElements(self):
+        return Seq(set(self))
```

Comment:
It doesn't really matter here, but I think it's the type dafny assumes.

Reply:
`UniqueElements` is just something that needs to be called in the target language, it doesn't have to produce a "Dafny value" at this level of abstraction. Just `set(self)` will work because it will only be used to compile something like this:

```dafny
var mySet := set x | x in mySeq
```

To something like this in Python:

```python
mySet = _dafny.Set()
for x in mySet.UniqueElements:  // <- produced by PythonCompiler.CreateForEachLoop
  mySet.add(x)
```

What you've got is correct too, since `_dafny.Seq` is also iterable, I'm just trying to save us an extra object creation. :)

Reply:
It will undoubtedly work for python(see it doesn’t matter _here_) and given that this is only used immediately in comprehensions, like you described, it doesn’t matter in general. That being said, I do like the cleaner interface of returning a sequence, but your overhead argument is compelling, so I’ll adopt it.

Reply:
Thanks! It's purely an optimization, and normally I wouldn't prioritize it at this stage, but this is a really easy one. :)